### PR TITLE
Adding cosine similarity based on all 3 majors, idea, and additional info

### PR DESCRIPTION
**Pull request by agrawal-prakhar**
This code pulls data from the CSV, renames the column headings, and then uses the data in the fields of major1, major2, major3, idea, and additional info to calculate the cosine similarity of all the different rows/entries in the CSV.

It is first concatenating these fields, and then calculating the cosine similarity. The result is a matrix of size [n X n] with each value representing the similarity between the row and column number entry.

### CODE REVIEW

File: matching_backend/matching_algo/cosine_similarity.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import pandas as pd
```
Comment (DhyeyMavani2003) [inline matching_backend/matching_algo/cosine_similarity.py:2]: Add a comment for import statements

**Review by DhyeyMavani2003 — CHANGES_REQUESTED** [review]
test: Please address comments

File: matching_backend/matching_algo/cosine_similarity.py (hunk 1 above)
Comment (agrawal-prakhar) [reply matching_backend/matching_algo/cosine_similarity.py:2]: Resolved

**Review by DhyeyMavani2003 — APPROVED** [review]
LGTM

**Pull request merged**